src/ucs/debug/log.c

```diff
@@ -94,34 +115,127 @@ void ucs_log_flush()
+
+            for (idx = start_idx; idx != UINT_MAX; --idx) {
+                ucs_assert(idx <= start_idx);
+
+                ucs_log_get_file_name(old_log_file_name,
+                                      UCS_LOG_MAX_FILE_NAME, idx);
+                ucs_snprintf_zero(new_log_file_name, UCS_LOG_MAX_FILE_NAME,
+                                  "%s.%u", ucs_log_file_names[idx], idx + 1);
+
+                ucs_assertv_always(access(old_log_file_name, W_OK) == 0,
+                                   "unable to write to %s", old_log_file_name);
+
+                /* coverity[toctou] */
+                ret = rename(old_log_file_name, new_log_file_name);
+                if (ret) {
+                    ucs_fatal("failed to rename %s to %s: %m",
+                              old_log_file_name, new_log_file_name);
+                }
+
+                ucs_assertv_always(access(old_log_file_name, F_OK) == -1,
+                                   "%s must not exist", old_log_file_name);
+                ucs_assertv_always(access(new_log_file_name, W_OK) == 0,
+                                   "unable to write to %s", new_log_file_name);
+
+                ucs_free(ucs_log_file_names[idx + 1]);
+                ucs_log_file_names[idx + 1] =
+                    ucs_strdup(ucs_log_file_names[idx], "backup logfile name");
+                if (ucs_log_file_names[idx + 1] == NULL) {
+                    ucs_fatal("unable to allocate name for backup log file #%u",
+                              idx + 1);
+                }
+            }
+        } else {
+            unlink(ucs_log_file_names[0]);
+        }
+
+        ucs_free(ucs_log_file_names[0]);
+        ucs_log_file_names[0] = NULL;
+        ucs_open_output_stream(ucs_global_opts.log_file,
+                               UCS_LOG_LEVEL_FATAL,
+                               &ucs_log_file, &ucs_log_file_close,
+                               &next_token, &ucs_log_file_names[0]);
+    }
 }
 
 static void ucs_log_print(size_t buffer_size, const char *short_file, int line,
                           ucs_log_level_t level,
                           const ucs_log_component_config_t *comp_conf,
                           const struct timeval *tv, const char *message)
 {
-    char *valg_buf;
+    char *log_buf;
+    int log_entry_len;
 
     if (RUNNING_ON_VALGRIND) {
-        valg_buf = ucs_alloca(buffer_size + 1);
-        snprintf(valg_buf, buffer_size,
-                 "[%lu.%06lu] %16s:%-4u %-4s %-5s %s\n", tv->tv_sec, tv->tv_usec,
-                 short_file, line, comp_conf->name,
-                 ucs_log_level_names[level], message);
-        VALGRIND_PRINTF("%s", valg_buf);
+        log_buf = ucs_alloca(buffer_size + 1);
+        snprintf(log_buf, buffer_size, UCS_LOG_SHORT_FMT,
+                UCS_LOG_SHORT_ARG(short_file, line, level,
+                                  comp_conf, tv, message));
+        VALGRIND_PRINTF("%s", log_buf);
```

Comment:
this format string cleanup should have been part of separate PR

Reply:
will upstream as separate PR

Reply:
#5120 - new PR
will rebase on top of it after merging